From bf08d6c5689d6fae4c4f2c4e35256cc01111050c Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Wed, 5 Feb 2025 17:13:07 +0100
Subject: [PATCH 1649/1695] drivers: stm32_exti: drop export of internal API

With the rework of EXTI driver as a hierarchical interrupt
controller and the rework of all the client drivers there is no
more any use for these API.

Drop the unused API.
Tag as static the API used internally.
Drop the include file stm32_exti.h

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 699238be361022c428aabd518fee367d96a6e236 drivers: stm32_exti: drop export of internal API]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: Icbdf6517196c38bc8443c695080d8a1acd4f83b2
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/447372
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/drivers/stm32_exti.c         | 55 ++++--------------
 core/include/drivers/stm32_exti.h | 93 -------------------------------
 2 files changed, 12 insertions(+), 136 deletions(-)
 delete mode 100644 core/include/drivers/stm32_exti.h

diff --git a/core/drivers/stm32_exti.c b/core/drivers/stm32_exti.c
index 3eaa4ddaa..8323b13ed 100644
--- a/core/drivers/stm32_exti.c
+++ b/core/drivers/stm32_exti.c
@@ -3,7 +3,6 @@
  * Copyright (c) 2021-2023, STMicroelectronics
  */
 
-#include <drivers/stm32_exti.h>
 #include <drivers/stm32_rif.h>
 #include <dt-bindings/interrupt-controller/irq.h>
 #include <io.h>
@@ -151,8 +150,8 @@ stm32_exti_event_is_configurable(const struct stm32_exti_pdata *exti,
 	return exti->trg[i] & mask;
 }
 
-void stm32_exti_set_type(struct stm32_exti_pdata *exti, uint32_t exti_line,
-			 uint32_t type)
+static void stm32_exti_set_type(struct stm32_exti_pdata *exti,
+				uint32_t exti_line, uint32_t type)
 {
 	uint32_t mask = BIT(exti_line % _EXTI_LINES_PER_BANK);
 	uint32_t r_trig = 0;
@@ -188,7 +187,7 @@ void stm32_exti_set_type(struct stm32_exti_pdata *exti, uint32_t exti_line,
 	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
-void stm32_exti_mask(struct stm32_exti_pdata *exti, uint32_t exti_line)
+static void stm32_exti_mask(struct stm32_exti_pdata *exti, uint32_t exti_line)
 {
 	uint32_t mask = BIT(exti_line % _EXTI_LINES_PER_BANK);
 	uint32_t exceptions = 0;
@@ -204,7 +203,8 @@ void stm32_exti_mask(struct stm32_exti_pdata *exti, uint32_t exti_line)
 	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
-void stm32_exti_unmask(struct stm32_exti_pdata *exti, uint32_t exti_line)
+static void stm32_exti_unmask(struct stm32_exti_pdata *exti,
+			      uint32_t exti_line)
 {
 	uint32_t mask = BIT(exti_line % _EXTI_LINES_PER_BANK);
 	uint32_t exceptions = 0;
@@ -220,7 +220,8 @@ void stm32_exti_unmask(struct stm32_exti_pdata *exti, uint32_t exti_line)
 	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
-void stm32_exti_enable_wake(struct stm32_exti_pdata *exti, uint32_t exti_line)
+static void stm32_exti_enable_wake(struct stm32_exti_pdata *exti,
+				   uint32_t exti_line)
 {
 	uint32_t mask = BIT(exti_line % _EXTI_LINES_PER_BANK);
 	uint32_t exceptions = 0;
@@ -235,7 +236,8 @@ void stm32_exti_enable_wake(struct stm32_exti_pdata *exti, uint32_t exti_line)
 	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
-void stm32_exti_disable_wake(struct stm32_exti_pdata *exti, uint32_t exti_line)
+static void stm32_exti_disable_wake(struct stm32_exti_pdata *exti,
+				    uint32_t exti_line)
 {
 	uint32_t mask = BIT(exti_line % _EXTI_LINES_PER_BANK);
 	uint32_t exceptions = 0;
@@ -250,7 +252,7 @@ void stm32_exti_disable_wake(struct stm32_exti_pdata *exti, uint32_t exti_line)
 	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
-void stm32_exti_clear(struct stm32_exti_pdata *exti, uint32_t exti_line)
+static void stm32_exti_clear(struct stm32_exti_pdata *exti, uint32_t exti_line)
 {
 	uint32_t mask = BIT(exti_line % _EXTI_LINES_PER_BANK);
 	uint32_t exceptions = 0;
@@ -266,7 +268,8 @@ void stm32_exti_clear(struct stm32_exti_pdata *exti, uint32_t exti_line)
 	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
-void stm32_exti_set_tz(struct stm32_exti_pdata *exti, uint32_t exti_line)
+static void stm32_exti_set_tz(struct stm32_exti_pdata *exti,
+			      uint32_t exti_line)
 {
 	uint32_t mask = BIT(exti_line % _EXTI_LINES_PER_BANK);
 	uint32_t exceptions = 0;
@@ -281,22 +284,6 @@ void stm32_exti_set_tz(struct stm32_exti_pdata *exti, uint32_t exti_line)
 	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
-void stm32_exti_set_gpio_port_sel(struct stm32_exti_pdata *exti, uint8_t bank,
-				  uint8_t pin)
-{
-	uint32_t reg = _EXTI_CR(pin / 4);
-	uint32_t shift = (pin % 4) * 8;
-	uint32_t val = bank << shift;
-	uint32_t mask = 0xff << shift;
-	uint32_t exceptions = 0;
-
-	exceptions = cpu_spin_lock_xsave(&exti->lock);
-
-	io_mask32(exti->base + reg, val, mask);
-
-	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
-}
-
 static struct itr_desc *
 stm32_exti_get_parent_itr(struct stm32_exti_pdata *exti, size_t it)
 {
@@ -693,24 +680,6 @@ static enum itr_return stm32_exti_it_handler(struct itr_handler *h)
 	return ITRR_HANDLED;
 }
 
-TEE_Result stm32_exti_get_pdata(const void *fdt, int nodeoffset,
-				struct stm32_exti_pdata **exti)
-{
-	struct itr_desc itr_desc = { };
-	TEE_Result res = TEE_ERROR_GENERIC;
-
-	res = dt_driver_device_from_node_idx_prop("wakeup-parent",
-						  fdt, nodeoffset, 0,
-						  DT_DRIVER_INTERRUPT,
-						  &itr_desc);
-	if (res)
-		return res;
-
-	*exti = itr_chip_to_stm32_exti_pdata(itr_desc.chip);
-
-	return TEE_SUCCESS;
-}
-
 /* Callback for "interrupts" and "interrupts-extended" DT node properties */
 static TEE_Result
 stm32_exti_dt_get_chip_cb(struct dt_pargs *pargs, void *priv_data,
diff --git a/core/include/drivers/stm32_exti.h b/core/include/drivers/stm32_exti.h
deleted file mode 100644
index e47276f29..000000000
--- a/core/include/drivers/stm32_exti.h
+++ /dev/null
@@ -1,93 +0,0 @@
-/* SPDX-License-Identifier: BSD-3-Clause */
-/*
- * Copyright (c) 2021, STMicroelectronics
- */
-
-#ifndef __STM32_EXTI_H
-#define __STM32_EXTI_H
-
-#include <tee_api_types.h>
-
-struct stm32_exti_pdata;
-
-/*
- * Recover the pointer to stm32_exti_pdata from DT property
- * "wakeup-parent"
- * @fdt: FDT base address
- * @nodeoffset: node offset in the FDT
- * @exti: the EXTI private data
- */
-#ifdef CFG_STM32_EXTI
-TEE_Result stm32_exti_get_pdata(const void *fdt, int nodeoffset,
-				struct stm32_exti_pdata **exti);
-#else
-static inline
-TEE_Result stm32_exti_get_pdata(const void *fdt __unused,
-				int nodeoffset __unused,
-				struct stm32_exti_pdata **exti __unused)
-{
-	return TEE_ERROR_NOT_IMPLEMENTED;
-}
-#endif
-
-/*
- * Set EXTI type in RTSR and FTSR EXTI registers
- * @exti_line: EXTI line number
- * @type: type (rising, falling or both) to set
- * Return a TEE_Result compliant return value
- */
-void stm32_exti_set_type(struct stm32_exti_pdata *exti, uint32_t exti_line,
-			 uint32_t type);
-
-/*
- * Mask EXTI Interrupt (IMR)
- * @exti_line: EXTI line number
- * Return a TEE_Result compliant return value
- */
-void stm32_exti_mask(struct stm32_exti_pdata *exti, uint32_t exti_line);
-
-/*
- * Unmask EXTI Interrupt (IMR)
- * @exti_line: EXTI line number
- * Return a TEE_Result compliant return value
- */
-void stm32_exti_unmask(struct stm32_exti_pdata *exti, uint32_t exti_line);
-
-/*
- * Enable EXTI line as wakeup interrupt
- * @exti_line: EXTI line number
- * Return a TEE_Result compliant return value
- */
-void stm32_exti_enable_wake(struct stm32_exti_pdata *exti, uint32_t exti_line);
-
-/*
- * Disable EXTI line as wakeup interrupt
- * @exti_line: EXTI line number
- * Return a TEE_Result compliant return value
- */
-void stm32_exti_disable_wake(struct stm32_exti_pdata *exti, uint32_t exti_line);
-
-/*
- * Clear pending EXTI interrupts
- * @exti_line: EXTI line number
- * Return a TEE_Result compliant return value
- */
-void stm32_exti_clear(struct stm32_exti_pdata *exti, uint32_t exti_line);
-
-/*
- * Configure EXTI mux for GPIO irq
- * @bank: GPIO bank id
- * @pin: GPIO number in the bank
- * Return a TEE_Result compliant return value
- */
-void stm32_exti_set_gpio_port_sel(struct stm32_exti_pdata *exti, uint8_t bank,
-				  uint8_t pin);
-
-/*
- * Securize the EXTI line
- * @exti_line: EXTI line number
- * Return a TEE_Result compliant return value
- */
-void stm32_exti_set_tz(struct stm32_exti_pdata *exti, uint32_t exti_line);
-
-#endif /*__STM32_EXTI_H*/
-- 
2.47.2

